File: Python/226_Invert_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if root == None:
            return None
        if root.left == None and root.right == None:
            return root
        else:
            a = None
            b = None
            if root.left != None:
                a = self.invertTree(root.left)
            if root.right != None:
                b = self.invertTree(root.right)
            root.left, root.right = b, a
            return root

File: C++/226_Invert_Binary_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if(root == NULL) return NULL;
        TreeNode* left = NULL;
        TreeNode* right = NULL;
        if(root->left != NULL){
            left = invertTree(root->left);
        }
        if(root->right != NULL){
            right = invertTree(root->right);
        }
        root->left = right;
        root->right = left;
        return root;
    }
};